### PR TITLE
Add Ability to Exlude Paths from Archive

### PR DESCRIPTION
**Pull request by rjsab**
##### ISSUE TYPE

- Feature Pull Request
##### COMPONENT NAME

files/archive.
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

The archive module allows you to pass a glob to the path argument.
This feature allows end users to continue to pass a glob to the path argument,
but enables a new argument to exclude specific dirs, or a glob of dirs in order to prevent
unwanted paths from being archived. The idea is to enable similar functionality as the
`--exclude=PATTERN` option for the `tar` command.

### CODE REVIEW

**Comment by rjsab** [comment]
Migrated from https://github.com/ansible/ansible-modules-extras/pull/3069

File: lib/ansible/modules/files/archive.py
Comment (bendoh) [inline lib/ansible/modules/files/archive.py]: Typo: `exluding` should be `excluding`

File: lib/ansible/modules/files/archive.py
Comment (bendoh) [inline lib/ansible/modules/files/archive.py]: Typo: `exlude` -> `exclude`

**Review by bendoh — CHANGES_REQUESTED** [review]
Looks good to me! There are a couple of minor typos that should be cleaned up.

**Comment by rjsab** [comment]
I have corrected the typos, thank you for catching them!

**Comment by rjsab** [comment]
Is there more that needs to be done to have this merged?

**Comment by abadyan-vonage** [comment]
@bendoh Is this expected to be merged into ansible? We also need this functionality.

**Comment by rjsab** [comment]
I've rebased with devel to fix the merge conflict. Let me know if there is more that you would like to see added in order to move the PR forward. I'd love to get this merged and closed off since it's been open for so long.

**Comment by bendoh** [comment]
shipit

**Comment by bendoh** [comment]
bot_status

**Comment by bendoh** [comment]
!needs_revision

I'm only marked as a contributor to this, so I don't believe I have any influence on the bot.

**Comment by bendoh** [comment]
!needs_revision

**Comment by rjsab** [comment]
@bendoh Does it need revision or are you trying to get the bot to remove the flag?

**Comment by bendoh** [comment]
AFAICT it doesn't need any revision, just trying to get the bot to undo the flag –– It LGTM tho!

**Comment by rjsab** [comment]
ready_for_review

**Comment by rjsabatini** [comment]
bot_broken

**Comment by mattclay** [comment]
Removing broken label to permit the bot to continue processing this PR. This PR appears to be waiting on approval from a second maintainer as documented here:

https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md#existing-modules

**Comment by rjsab** [comment]
@mattclay Who would the second maintainer be? I only know of @bendoh. I added the bot broken label because it wasn't properly removing the needs revision tag.

**Comment by mattclay** [comment]
A second approval from either a maintainer in the same namespace (files) or from a core maintainer should be adequate. @jctanner Is that correct?

**Comment by abadger** [comment]
+1 from me too.  WIll merge to devel

**Comment by abadger** [comment]
Merged to devel for the 2.4.0 release.  Thanks @rjsabatini for the code and @bendoh for the review!

**Comment by mkrizek** [comment]
@bendoh Your `!needs_revision` command didn't work because your review still says that you requested changes, you'd need to approve the PR via gihub UI to get rid of that label.

**Pull request merged**
